src/flowcean/polars/transforms/discrete_derivative.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from collections.abc import Iterable
+from typing import Literal, TypeAlias
+
+import polars as pl
+from typing_extensions import override
+
+from flowcean.core import Transform
+
+logger = logging.getLogger(__name__)
+
+DiscreteDerivativeType: TypeAlias = Literal["forward", "backward", "central"]
+
+
+class DiscreteDerivative(Transform):
+    """Calculates the discrete derivative of time series features.
+
+    Calculates the discrete derivative of time series features using either
+    forward, backward, or central differences.
+    """
+
+    def __init__(
+        self,
+        features: str | Iterable[str],
+        *,
+        method: DiscreteDerivativeType = "central",
+    ) -> None:
+        """Initializes the DiscreteDerivative transform.
+
+        Args:
+            features: Features that shall be differentiated. Result features
+                will be named `<feature>_derivative`.
+            method: Method to use for calculating the derivative. Valid options
+                are "forward", "backward", and "central".
+                Defaults to "central".
+        """
+        self.features = [features] if isinstance(features, str) else features
+        self.method = method
+
+    @override
+    def apply(self, data: pl.LazyFrame) -> pl.LazyFrame:
+        feature_names = data.collect_schema().names()
+        for feature in self.features:
+            value_feature = f"{feature}_value"
+            time_feature = f"{feature}_t"
+            dt_feature = f"{feature}_dt"
+            dvalue_feature = f"{feature}_derivative"
+            index_feature = "_index"
+
+            data = (
+                data.with_columns(
+                    [
+                        pl.col(feature)
+                        .list.eval(pl.first().struct.field("value"))
+                        .alias(value_feature),
+                        pl.col(feature)
+                        .list.eval(pl.first().struct.field("time"))
+                        .alias(time_feature),
+                    ],
+                )
+                .with_row_index(name=index_feature)
+                .explode([value_feature, time_feature])
```

Comment:
this still contains all the features in the data frame, does it? Can we select beforehand? only using the features for the derivative?